components/resource_control/src/resource_group.rs

```diff
@@ -383,14 +383,55 @@ impl ResourceController {
             CommandPri::Normal => 1,
             CommandPri::High => 0,
         };
-        self.resource_group(name).get_priority(level)
+        self.resource_group(name).get_priority(level, None)
+    }
+}
+
+pub struct TaskMetadata {
+    // the resource group name of this task.
+    pub group_name: String,
+    // the override priority of this task neglecting the original resource group priority.
+    pub override_priority: u32,
+}
+
+impl TaskMetadata {
+    pub fn to_vec(self) -> Vec<u8> {
+        let mut buf = Vec::with_capacity(4 + self.group_name.len());
```

Comment:
NIT: maybe we can avoid an extra malloc by reusing the memory of group_name if it already has the extra 4 bytes.